### DISCUSSION TITLE
Dual Vacuum & Dual Camera Support

### DISCUSSION THREAD

**Discussion by scarylukas** [sca075/mqtt_vacuum_camera#267] [General]
Hello
I have two Xiaomi 1C devices with ValeTudo support.

Integration MQTT Vacuum's Camera in my case is only able to handle one camera correctly. When I try to add a second camera (another entity) I get the message "already_configured". Is it possible to handle two cameras from two different devices?

**Comment by sca075** [comment]
Sure but you should rename one of the two vacuums. Normally Valetudo names the vacuums per model.
When the vacuums have a different "names" identifiers it should be possible to add the second camera. Or you can simply rename the camera entry id of the first and it should work fine too.

**Comment by scarylukas** [comment]
The names are different.

Vaccum1 - System ID "PleasedMutedWolf", Custom Friendly Name "AutoVacuum - 1Floor", 
Vacuum2 - System ID "HandyTornDeer", Custom Friendly Name "AutoVacuum - 0Floor",

Changing the names to other random ones does not help. 

Adding first vacuum
![Screenshot 2024-11-12 at 08 54 48](https://github.com/user-attachments/assets/3df0b62b-6f91-4e12-9d5b-71e3b35bb850)
![Screenshot 2024-11-12 at 08 54 59](https://github.com/user-attachments/assets/a778ddd2-4816-4b75-a472-4af2ec0efb37)

Adding second vacuum
![Screenshot 2024-11-12 at 08 55 34](https://github.com/user-attachments/assets/2a8629d5-cb21-46ba-825d-2dfaa086bab2)
![Screenshot 2024-11-12 at 09 10 16](https://github.com/user-attachments/assets/3b756dd1-0a9d-4039-890e-81707b5c703c)
![Screenshot 2024-11-12 at 08 55 44](https://github.com/user-attachments/assets/d5c534c5-151e-4f97-84c2-733a31c5fea7)

**Reply by sca075** [reply]
@scarylukas I think I got the problem the  is the camera name.. It could be that "-" under the name of the vacuum is creating the same camera entry_id.. can you please confirm if by changing the camera entity_id the problem is fix (honestly I taught this was fixed in one of the previous versions, if not I will move the discussion on the issues and I will try to fix it. 

Thanks a lot for letting me know that ;)

**Comment by scarylukas** [comment]
@sca075 changing the camera entity_ID does not solve the problem

**Reply by sca075** [reply]
@scarylukas can you please show me the camera entity I mean it should be something  camera.area38_0_floor_vacuum_camera is that correct? or the camera.area38_camera??  think this "_" is the problem.  if the vacuum had  vacuum.area38floor0_vacuum then you should have both vacuums as per I just tested that the vacuum.vacuum_rockrobo generate camera.vacuum.. I repeat this is the problem here... and as promised I'm going to fix it as soon. But as work arround..

![Screenshot 2024-11-12 at 22 30 37](https://github.com/user-attachments/assets/41cf7e6f-2fe3-413d-9d58-7e33ca6d8f53)

![Screenshot 2024-11-12 at 22 38 07](https://github.com/user-attachments/assets/8195c0fb-7893-489d-a474-df82f782b995)

**Reply by scarylukas** [reply]
![Screenshot 2024-11-12 at 23 05 28](https://github.com/user-attachments/assets/e873e02c-42eb-4be6-a99f-f6902157f69a)
![Screenshot 2024-11-12 at 23 06 55](https://github.com/user-attachments/assets/0321119c-024b-4336-a1a3-a940eedb6c28)

**Reply by sca075** [reply]
Did you configured the second camera too?

**Comment by sca075** [comment]
@scarylukas  just opened also the issue #268 for follow-up

**Reply by sca075** [reply]
@scarylukas I did some homework and I think the cause is that both vacuums have the "same topic" this means:  
autovacuum - 1Floor = autovacuum - 0Floor so when I change the logic this will impact several users.
![Screenshot 2024-11-17 at 22 16 16](https://github.com/user-attachments/assets/be788b87-ff84-4588-9a0c-2d83cd580460)

The camera use the data in this fields to generate the unique_id. Can you be so kind to check and eventually change those value uniquely for both vacuums this should allow you to configure both vacuums.

**Reply by scarylukas** [reply]
@sca075 I confirm. After changing the topic to a unique one I can add two vacuums. Thank you very much for your help
